### PR TITLE
Support file downloading via SFTP

### PR DESCRIPTION
**Pull request by tcamin**
This PR adds download support of files via SFTP as a further improvement of what has been done in #15

### CODE REVIEW

File: Sources/Bindings/SFTP.swift [hunk 1]
```diff
@@ -61,6 +65,38 @@ public class SFTP {
...
+            codeOrBytesReceived = libssh2_sftp_read(sftpHandle, &buffer, SFTP.bufferSize)
+            if codeOrBytesReceived < 0 && codeOrBytesReceived != Int(LIBSSH2_ERROR_EAGAIN) {
+                throw LibSSH2Error(code: Int32(codeOrBytesReceived), message: "libssh2_sftp_read failed")
+            }
+            
+            let data = Data(buffer: UnsafeBufferPointer(start: &buffer, count: codeOrBytesReceived))
+            fileHandle.write(data)
+        } while codeOrBytesReceived > 0
+        
+        libssh2_sftp_close_handle(sftpHandle)
```
Comment (jakeheis) [inline Sources/Bindings/SFTP.swift]: Should this be in a `defer` so that the handle is still closed even if an error is thrown?

**Comment by jakeheis** [comment]
Looks good to me, just left a single comment!

File: Sources/Bindings/SFTP.swift (hunk 1 above)
Comment (tcamin) [reply Sources/Bindings/SFTP.swift]: Fixed. BTW I found that the same issue was in the upload() method as well, I've fixed it in 896c74b

**Comment by jakeheis** [comment]
Thanks for the contribution!

**Pull request merged**